### PR TITLE
Chapter 1

### PR DESCRIPTION
**Pull request by mridul2899**
### Solutions for Chapter 1

cc @vrom911 @chshersh
Sorry for the previous pull request with incorrect code. I've rectified the errors for this one. Kindly review.

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -652,8 +670,13 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n
+    | absN == 0 = 0
+    | absN < 10 = absN
```
Comment (vrom911) [inline src/Chapter1.hs:676]: Zero cas is covered by `< 10` for the absolute value as well, so it is safe to remove this case:

```suggestion
    | absN < 10 = absN
```

File: src/Chapter1.hs [hunk 2]
```diff
@@ -652,8 +670,13 @@ Implement a function that returns the first digit of a given number.
...
+    | otherwise = firstDigit (div absN 10)
+      where
+        absN = abs n
```
Comment (vrom911) [inline src/Chapter1.hs:679]: Great usage of the `where` construction 👏🏼

**Review by vrom911 — APPROVED** [review]
Awesome work on solving the challenges! ⭐ 
Congrats with finishing Chapter One 🥳 

And well done with the Advanced Task :medal_sports: 

I left my review and some comments. Hope you find them helpful :slightly_smiling_face:

**Comment by mridul2899** [comment]
Thank you very much for reviewing the code and giving insightful feedback. I found your suggestions very helpful and have made the appropriate changes.

**Comment by vrom911** [comment]
@mridul2899 looks fantastic! 🤩 
 I'm going to close this PR, because we can't merge the solutions to our repo :slightly_smiling_face: But don't worry, it still counts towards Hacktoberfest contribution :shirt: :deciduous_tree: Feel free to open PRs with solutions to other chapters!

**Pull request closed without merging**